File: vben-admin/apps/web-antd/src/api/auxiliary.ts
import { requestClient } from '#/api/request';

// ===== 卡密系统 =====
export interface CardKey {
  id: number;
  content: string;
  money: number;
  status: number;
  uid: number | null;
  addtime: string;
  usedtime: string;
}

export interface CardKeyListResult {
  list: CardKey[];
  pagination: { page: number; limit: number; total: number };
}

export async function getCardKeyListApi(params: { page?: number; limit?: number; status?: number } = {}) {
  return requestClient.get<CardKeyListResult>('/admin/cardkeys', { params });
}

export async function generateCardKeysApi(money: number, count: number) {
  return requestClient.post<{ codes: string[]; count: number }>('/admin/cardkey/generate', { money, count });
}

export async function deleteCardKeysApi(ids: number[]) {
  return requestClient.post<{ deleted: number }>('/admin/cardkey/delete', { ids });
}

export async function useCardKeyApi(content: string) {
  return requestClient.post<{ money: number; msg: string }>('/user/cardkey/use', { content });
}

// ===== 活动系统 =====
export interface Activity {
  hid: number;
  name: string;
  yaoqiu: string;
  type: string;
  num: string;
  money: string;
  addtime: string;
  endtime: string;
  status_ok: string;
  status: string;
}

export interface ActivityListResult {
  list: Activity[];
  pagination: { page: number; limit: number; total: number };
}

export async function getActivityListApi(params: { page?: number; limit?: number } = {}) {
  return requestClient.get<ActivityListResult>('/admin/activities', { params });
}

export async function saveActivityApi(data: Partial<Activity>) {
  return requestClient.post('/admin/activity/save', data);
}

export async function deleteActivityApi(hid: number) {
  return requestClient.delete(`/admin/activity/${hid}`);
}

export async function getPublicActivityListApi() {
  return requestClient.get<Activity[]>('/activities');
}

// ===== 质押系统 =====
export interface PledgeConfig {
  id: number;
  category_id: number;
  amount: number;
  discount_rate: number;
  status: number;
  addtime: string;
  days: number;
  cancel_fee: number;
  category_name?: string;
}

export interface PledgeRecord {
  id: number;
  uid: number;
  config_id: number;
  status: number;
  addtime: string;
  endtime: string | null;
  amount?: number;
  category_name?: string;
  discount_rate?: number;
  days?: number;
  username?: string;
}

export interface PledgeRecordListResult {
  list: PledgeRecord[];
  pagination: { page: number; limit: number; total: number };
}

// 管理端
export async function getPledgeConfigListApi() {
  return requestClient.get<PledgeConfig[]>('/admin/pledge/configs');
}

export async function savePledgeConfigApi(data: Partial<PledgeConfig>) {
  return requestClient.post('/admin/pledge/config/save', data);
}

export async function deletePledgeConfigApi(id: number) {
  return requestClient.delete(`/admin/pledge/config/${id}`);
}

export async function togglePledgeConfigApi(id: number, status: number) {
  return requestClient.post('/admin/pledge/config/toggle', { id, status });
}

export async function getPledgeRecordListApi(params: { page?: number; limit?: number; uid?: number } = {}) {
  return requestClient.get<PledgeRecordListResult>('/admin/pledge/records', { params });
}

// 用户端
export async function getUserPledgeConfigsApi() {
  return requestClient.get<PledgeConfig[]>('/pledge/configs');
}

export async function createPledgeApi(config_id: number) {
  return requestClient.post('/pledge/create', { config_id });
}

export async function cancelPledgeApi(id: number) {
  return requestClient.post(`/pledge/cancel/${id}`);
}

export async function getMyPledgesApi() {
  return requestClient.get<PledgeRecord[]>('/pledge/my');
}

// ===== 外部查单 =====
export interface CheckOrderResult {
  oid: number;
  ptname: string;
  kcname: string;
  status: string;
  process: string;
  remarks: string;
  addtime: string;
  account?: string;
  school?: string;
  pushUid?: string;
  pushStatus?: string;
  pushEmail?: string;
  pushEmailStatus?: string;
  showdoc_push_url?: string;
  pushShowdocStatus?: string;
}

export async function checkOrderApi(params: { user?: string; oid?: string }) {
  return requestClient.get<{ list: CheckOrderResult[]; total: number }>('/query', { params });
}

// ===== 消息推送与自动登录 =====

export async function pushBindWxApi(data: { account: string; pushUid: string; oids: string }) {
  return requestClient.post('/push/bind-wx', data);
}

export async function pushUnbindWxApi(data: { account: string }) {
  return requestClient.post('/push/unbind-wx', data);
}

export async function pushBindEmailApi(data: { orderid?: number; account?: string; pushEmail: string }) {
  return requestClient.post('/push/bind-email', data);
}

export async function pushUnbindEmailApi(data: { orderid?: number; account?: string }) {
  return requestClient.post('/push/unbind-email', data);
}

export async function pushBindShowDocApi(data: { orderid?: number; account?: string; showdoc_url: string }) {
  return requestClient.post('/push/bind-showdoc', data);
}

export async function pushUnbindShowDocApi(data: { orderid?: number; account?: string }) {
  return requestClient.post('/push/unbind-showdoc', data);
}

export async function pushWxQrcodeApi(data: { account: string }) {
  return requestClient.post<{ code: string; url: string; shortUrl: string }>('/push/wx-qrcode', data);
}

export async function pushWxScanUidApi(data: { code: string }) {
  return requestClient.post<{ uid: string }>('/push/wx-scan-uid', data);
}

export async function pushPupLoginApi(oid: number) {
  return requestClient.get<{ url: string }>('/push/puplogin', { params: { oid } });
}

File: go-api/internal/model/auxiliary.go
package model

import "time"

// ===== 卡密系统 =====

type CardKey struct {
	ID       int    `json:"id" db:"id"`
	Content  string `json:"content" db:"content"`
	Money    int    `json:"money" db:"money"`
	Status   int    `json:"status" db:"status"`
	UID      *int   `json:"uid" db:"uid"`
	AddTime  string `json:"addtime" db:"addtime"`
	UsedTime string `json:"usedtime" db:"usedtime"`
}

type CardKeyGenRequest struct {
	Money int `json:"money" binding:"required,min=1"`
	Count int `json:"count" binding:"required,min=1,max=100"`
}

type CardKeyListRequest struct {
	Page   int `form:"page" json:"page"`
	Limit  int `form:"limit" json:"limit"`
	Status int `form:"status" json:"status" binding:"-"`
}

type CardKeyUseRequest struct {
	Content string `json:"content" binding:"required"`
}

// 等级系统的 Grade / GradeSaveRequest 已在 admin.go 中定义

// ===== 活动系统 =====

type Activity struct {
	HID      int    `json:"hid" db:"hid"`
	Name     string `json:"name" db:"name"`
	YaoQiu   string `json:"yaoqiu" db:"yaoqiu"`
	Type     string `json:"type" db:"type"`
	Num      string `json:"num" db:"num"`
	Money    string `json:"money" db:"money"`
	AddTime  string `json:"addtime" db:"addtime"`
	EndTime  string `json:"endtime" db:"endtime"`
	StatusOK string `json:"status_ok" db:"status_ok"`
	Status   string `json:"status" db:"status"`
}

type ActivitySaveRequest struct {
	HID      int    `json:"hid"`
	Name     string `json:"name" binding:"required"`
	YaoQiu   string `json:"yaoqiu" binding:"required"`
	Type     string `json:"type" binding:"required"`
	Num      string `json:"num" binding:"required"`
	Money    string `json:"money" binding:"required"`
	AddTime  string `json:"addtime" binding:"required"`
	EndTime  string `json:"endtime" binding:"required"`
	StatusOK string `json:"status_ok"`
}

type ActivityListRequest struct {
	Page  int `form:"page" json:"page"`
	Limit int `form:"limit" json:"limit"`
}

type ActivityRecord struct {
	ID       int       `json:"id" db:"id"`
	HID      int       `json:"hid" db:"hid"`
	UID      int       `json:"uid" db:"uid"`
	Progress int       `json:"progress" db:"progress"`
	Status   int       `json:"status" db:"status"`
	AddTime  time.Time `json:"addtime" db:"addtime"`
}

// ===== 质押系统 =====

type PledgeConfig struct {
	ID           int     `json:"id" db:"id"`
	CategoryID   int     `json:"category_id" db:"category_id"`
	Amount       float64 `json:"amount" db:"amount"`
	DiscountRate float64 `json:"discount_rate" db:"discount_rate"`
	Status       int     `json:"status" db:"status"`
	AddTime      string  `json:"addtime" db:"addtime"`
	Days         int     `json:"days" db:"days"`
	CancelFee    float64 `json:"cancel_fee" db:"cancel_fee"`
	CategoryName string  `json:"category_name,omitempty"`
}

type PledgeConfigSaveRequest struct {
	ID           int     `json:"id"`
	CategoryID   int     `json:"category_id" binding:"required"`
	Amount       float64 `json:"amount" binding:"required"`
	DiscountRate float64 `json:"discount_rate" binding:"required"`
	Days         int     `json:"days" binding:"required,min=1"`
	CancelFee    float64 `json:"cancel_fee"`
}

type PledgeRecord struct {
	ID       int     `json:"id" db:"id"`
	UID      int     `json:"uid" db:"uid"`
	ConfigID int     `json:"config_id" db:"config_id"`
	Status   int     `json:"status" db:"status"`
	AddTime  string  `json:"addtime" db:"addtime"`
	EndTime  *string `json:"endtime" db:"endtime"`
	Amount   float64 `json:"amount,omitempty"`
	CatName  string  `json:"category_name,omitempty"`
	Discount float64 `json:"discount_rate,omitempty"`
	Days     int     `json:"days,omitempty"`
	Username string  `json:"username,omitempty"`
}

type PledgeListRequest struct {
	Page  int `form:"page" json:"page"`
	Limit int `form:"limit" json:"limit"`
	UID   int `form:"uid" json:"uid"`
}

// ===== 网签系统 =====

type MlsxCompany struct {
	ID      int    `json:"id" db:"id"`
	QYMC    string `json:"qymc" db:"qymc"`
	WQBS    string `json:"wqbs" db:"wqbs"`
	ShiJian string `json:"shijian" db:"shijian"`
}

type MlsxCompanySaveRequest struct {
	ID   int    `json:"id"`
	QYMC string `json:"qymc" binding:"required"`
	WQBS string `json:"wqbs"`
}

type MlsxFile struct {
	ID      int    `json:"id" db:"id"`
	WJID    string `json:"wjid" db:"wjid"`
	Name    string `json:"name" db:"name"`
	IP      string `json:"ip" db:"ip"`
	ShiJian string `json:"shijian" db:"shijian"`
}

// ===== 外部查单 =====

type CheckOrderRequest struct {
	User   string `form:"user" json:"user"`
	OID    string `form:"oid" json:"oid"`
	KCName string `form:"kcname" json:"kcname"`
}

type CheckOrderResult struct {
	OID     int    `json:"oid"`
	PtName  string `json:"ptname"`
	KCName  string `json:"kcname"`
	Status  string `json:"status"`
	Process string `json:"process"`
	Remarks string `json:"remarks"`
	AddTime string `json:"addtime"`
}
